File: CS/DxBlazorApplication1/Components/MDI/MDITabCollection.cs
using DevExpress.Blazor;
using System.Collections;
using System.Text.Json.Serialization;

namespace DxBlazorApplication1.Components.MDI
{
    public class MDITabCollection {
        [JsonInclude]
        private List<MDITab> tabs;

        public int Count => tabs.Count;

        public MDITabCollection() {
            tabs = new List<MDITab>();
        }

        public MDITabCollection(IEnumerable<ITabInfo> tabs) {
            this.tabs = tabs.Select(t => new MDITab(t.Text,  
                t.VisibleIndex == -1 ? tabs.ToList().IndexOf(t) : t.VisibleIndex,
                t.Visible)).ToList();
        }

        public void SetVisibleAllTabs(bool visible) {
            tabs.ForEach((t) => t.Visible = visible);
        }

        public string? GetTabTextByTabInfo(ITabInfo tabInfo) {
            return tabs.FirstOrDefault(t => t.Text == tabInfo.Text)?.Text;
        }

        public int GetVisibleIndexByTabText(string? text) {
            return tabs.Find(t => t.Text == text)?.VisibleIndex ?? -1;
        }

        public bool GetVisibleByTabText(string? text) {
            return tabs.Find(t => t.Text == text)?.Visible ?? true;
        }

        public void SetVisibleByTabText(string? text, bool visible) {
            var tab = tabs.Find(t => t.Text == text);
            if(tab != null)
            {
                tab.Visible = visible;
            }
        }

        public void SetVisibleIndexByTabText(string? text, int visibleIndex) {
            var tab = tabs.Find(t => t.Text == text);
            if(tab != null)
            {
                tab.VisibleIndex = visibleIndex;
            }
        }
    }
}

File: VB/DxBlazorApplication1/Components/MDI/MDITabCollection.vb
Imports DevExpress.Blazor
Imports System.Collections
Imports System.Text.Json.Serialization

Namespace DxBlazorApplication1.Components.MDI

    Public Class MDITabCollection

        <JsonInclude>
        Private tabs As List(Of MDITab)

        Public ReadOnly Property Count As Integer
            Get
                Return tabs.Count
            End Get
        End Property

        Public Sub New()
            tabs = New List(Of MDITab)()
        End Sub

        Public Sub New(ByVal tabs As IEnumerable(Of ITabInfo))
            Me.tabs = tabs.[Select](Function(t) New MDITab(t.Text, If(t.VisibleIndex Is -1, tabs.ToList().IndexOf(t), t.VisibleIndex), t.Visible)).ToList()
        End Sub

        Public Sub SetVisibleAllTabs(ByVal visible As Boolean)
            tabs.ForEach(Function(t) CSharpImpl.__Assign(t.Visible, visible))
        End Sub

        Public Function GetTabTextByTabInfo(ByVal tabInfo As ITabInfo) As String?
            Return tabs.FirstOrDefault(Function(t) t.Text Is tabInfo.Text)?.Text
        End Function

        Public Function GetVisibleIndexByTabText(ByVal text As String?) As Integer
            Return If(tabs.Find(Function(t) t.Text Is text)?.VisibleIndex, -1)
        End Function

        Public Function GetVisibleByTabText(ByVal text As String?) As Boolean
            Return If(tabs.Find(Function(t) t.Text Is text)?.Visible, True)
        End Function

        Public Sub SetVisibleByTabText(ByVal text As String?, ByVal visible As Boolean)
            Dim tab = tabs.Find(Function(t) t.Text Is text)
            If tab IsNot Nothing Then
                tab.Visible = visible
            End If
        End Sub

        Public Sub SetVisibleIndexByTabText(ByVal text As String?, ByVal visibleIndex As Integer)
            Dim tab = tabs.Find(Function(t) t.Text Is text)
            If tab IsNot Nothing Then
                tab.VisibleIndex = visibleIndex
            End If
        End Sub

        Private Class CSharpImpl

            <System.Obsolete("Please refactor calling code to use normal Visual Basic assignment")>
            Shared Function __Assign(Of T)(ByRef target As T, value As T) As T
                target = value
                Return value
            End Function
        End Class
    End Class
End Namespace
